src/background.ts

```diff
@@ -30,24 +38,54 @@ const subscribe = (
     batching: { maxMilliseconds: 10 },
   });
   const childLogger = logger.child({ subscription });
-  sub.on('message', async (message) => {
-    try {
-      await handler(message, connection, childLogger, pubsub);
-      message.ack();
-    } catch (err) {
-      childLogger.error(
-        { messageId: message.id, data: message.data.toString('utf-8'), err },
-        'failed to process message',
-      );
-      message.nack();
-    }
+  const histogram = meter.createHistogram('message_processing_time', {
+    unit: 'ms',
+    description: 'time to process a message',
   });
+  sub.on('message', async (message) =>
+    runInRootSpan(
+      `message: ${subscription}`,
+      async (span) => {
+        const startTime = Date.now();
+        let success = true;
+        addPubsubSpanLabels(span, subscription, message);
```

Comment:
Once either https://github.com/googleapis/nodejs-pubsub/pull/1833 or https://github.com/googleapis/nodejs-pubsub/pull/1659 is merged, we could probably remove that, as it would automagically add the span labels itself :)

Reply:
Consider non blocking, but please add a backlog task for that.

Reply:
WT-1903